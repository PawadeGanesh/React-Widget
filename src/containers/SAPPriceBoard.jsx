import React, { Component } from 'react';
import axios from 'axios';
import moment from 'moment';
import { Table } from 'reactstrap';
import AlertBox from '../components/layout/AlertBox';

class SAPPriceBoard extends Component {
	state = {
		isLoading: false,
		data: [],
		page: 1,
		pageSize: 4
	};

	getSAPList() {
		this.setState({ isLoading: true }, () => {
			axios
				.get(`${process.env.REACT_APP_SAPPRICEMODELBASEURI}`)
				.then(res => {
					const data = res.data.contracts;
					this.setState({ isLoading: false, data: res.data.contracts });
					this.props.checkdata(data);
				})
				.catch(error => {
					this.setState({ isLoading: false, error });
				});
		});
	}

	componentDidMount() {
		this.getSAPList();
	}

	columns() {
		return [
			{
				name: 'supplierName',
				label: 'Supplier'
			},
			{
				name: 'rawMaterialName',
				label: 'Materials'
			},
			{
				name: 'country',
				label: 'Country'
			},
			{
				name: 'price',
				label: 'Price(in $)'
			},
			{
				name: 'dateCreated',
				label: 'Created Date ',
				onRender: row =>
					row.dateCreated ? moment(row.dateCreated).format('L') : '---'
			}
		];
	}

	onPageChange(page) {
		this.setState({ page });
	}

	render() {
		const { data, page, pageSize, isLoading } = this.state;

		const pages = [];

		const results = [...data];
		while (results.length > 0) pages.push(results.splice(0, pageSize));

		const fields = this.columns();
		return (
			<div>
				{isLoading ? (
					<div
						className='d-flex justify-content-center'
						style={{ marginTop: '10%' }}
					>
						<span className='spinner-border'></span>
					</div>
				) : (
					<div className='table-responsive'>
						{data && data.length === 0 ? (
							<AlertBox />
						) : (
							<Table
								style={{
									fontSize: '10px'
								}}
							>
								<thead>
									<tr style={{ backgroundColor: '#d5ebf0' }}>
										{fields.map(field => (
											<th
												className='white text-center'
												onClick={() => this.sort(field.name)}
											>
												{field.label}
											</th>
										))}
									</tr>
								</thead>
								<tbody>
									{pages[page - 1] &&
										pages[page - 1].map((eachRow, index) => {
											return (
												<tr key={index}>
													{fields.map(field => (
														<td className='white text-center'>
															{field.onRender
																? field.onRender(eachRow, index)
																: eachRow[field.name]}
														</td>
													))}
												</tr>
											);
										})}
								</tbody>
							</Table>
						)}
					</div>
				)}
			</div>
		);
	}
}

export default SAPPriceBoard;
