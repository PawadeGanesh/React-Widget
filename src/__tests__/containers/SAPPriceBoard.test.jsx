// import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import SAPPriceBoard from '../../containers/SAPPriceBoard';

describe('SAPPriceBoard', () => {
	it('should render correctly', () => {
		const wrapper = mount(<SAPPriceBoard />);
		expect(wrapper.render()).toMatchSnapshot();
	});

	it('renders in table rows based on provided columns', () => {
		const cols = [
			{ name: 'supplierName', label: 'Supplier' },
			{ name: 'rawMaterialName', label: 'Materials' },
			{ name: 'country', label: 'Country' },
			{ name: 'price', label: 'Price(in $)' }
		];

		const data = [
			{ name: 'John', label: 'P&G' },
			{ name: 'ABCD', label: 'P&G' },
			{ name: 'India', label: 'India' }
		];

		const container = mount(<SAPPriceBoard data={data} cols={cols} />);
		expect(container).toMatchSnapshot();
	});

	it('should render a loader', () => {
		const wrapper = mount(<SAPPriceBoard />);
		expect(wrapper.find('.spinner-border')).toMatchSnapshot();
	});

	it('should render a table', () => {
		const wrapper = mount(<SAPPriceBoard />);
		expect(wrapper.find('Table')).toMatchSnapshot();
	});

	it('should have -', () => {
		const wrapper = mount(<SAPPriceBoard />);
		expect(wrapper.find('-')).toMatchSnapshot();
	});
});
