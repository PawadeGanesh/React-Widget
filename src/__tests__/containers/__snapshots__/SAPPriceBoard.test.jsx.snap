// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAPPriceBoard renders in table rows based on provided columns 1`] = `ReactWrapper {}`;

exports[`SAPPriceBoard should render a loader 1`] = `ReactWrapper {}`;

exports[`SAPPriceBoard should render a table 1`] = `ReactWrapper {}`;

exports[`SAPPriceBoard should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "d-flex justify-content-center",
          "style": "margin-top: 10%;",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "spinner-border",
            },
            "children": Array [],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
